elide-core/src/main/java/com/yahoo/elide/core/security/visitors/InmemoryPermissionExpressionVisitor.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2021, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.core.security.visitors;
+
+import com.yahoo.elide.core.dictionary.EntityDictionary;
+import com.yahoo.elide.core.security.RequestScope;
+import com.yahoo.elide.core.security.checks.Check;
+import com.yahoo.elide.core.security.checks.DatastoreEvalFilterExpressionCheck;
+import com.yahoo.elide.core.security.checks.FilterExpressionCheck;
+import com.yahoo.elide.core.security.checks.UserCheck;
+import com.yahoo.elide.core.security.permissions.expressions.AndExpression;
+import com.yahoo.elide.core.security.permissions.expressions.CheckExpression;
+import com.yahoo.elide.core.security.permissions.expressions.Expression;
+import com.yahoo.elide.core.security.permissions.expressions.ExpressionVisitor;
+import com.yahoo.elide.core.security.permissions.expressions.NotExpression;
+import com.yahoo.elide.core.security.permissions.expressions.OrExpression;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Traverse the Expression Parser and returns the Permission Expression and boolean Pair.
+ */
+public class InmemoryPermissionExpressionVisitor implements ExpressionVisitor<Pair<Expression, Boolean>> {
```

Comment:
I'm not sure we need this boolean on the right.  Each individual expression should know if it needs to run in memory or not.

Reply:
If we don't have a boolean, then we have to include the executeInMemory or executeInDatastore variable in Check Expression. Then change it for all types of checks in the Check expression.